### PR TITLE
Update ruamel.yaml to >=0.18

### PR DESCRIPTION
**Pull request by trappitsch**
Part of #406 

- Fix configuration to use the new style
- Fix test where old loader was called

Currently I'm using the `unsafe` loader, which would be okay since the docs specify that this is not safe for files you don't trust. 

However, we could try and register all `ik` classes with the yaml loader, [see here](https://yaml.readthedocs.io/en/latest/dumpcls/). Scratching my head at the moment though why I can't even get the tests to run when I register `instruments.Instrument` with 

```python
from instruments.config import yaml

@yaml.register_class
class Instrument
    ...
```

Any ideas?

### CODE REVIEW

File: src/instruments/config.py [hunk 1]
```diff
@@ -8,17 +8,7 @@
...
-    import ruamel.yaml as yaml
-except ImportError:
-    # Some versions of ruamel.yaml are named ruamel_yaml, so try that
-    # too.
-    #
-    # In either case, we've observed issues with pylint where it will raise
-    # a false positive from its import-error checker, so we locally disable
-    # it here. Once the cause for the false positive has been identified,
-    # the import-error check should be re-enabled.
-    import ruamel_yaml as yaml  # pylint: disable=import-error
```
Comment (trappitsch) [inline src/instruments/config.py:21]: I don't see the point of this check if we pin `ruamel.yaml>=0.18`

File: src/instruments/config.py [hunk 2]
```diff
@@ -139,16 +130,11 @@ def load_instruments(conf_file_name, conf_path="/"):
         such processing will occur independently of the value of ``conf_path``.
     """
 
-    if yaml is None:
-        raise ImportError(
-            "Could not import ruamel.yaml, which is required " "for this function."
-        )
-
```
Comment (trappitsch) [inline src/instruments/config.py:146]: `ruamel.yaml` is not an optional dependency. So I see no need for this check... unless we make it an optional dependency ;)

**Comment by scasagrande** [comment]
> Any ideas?

Nope, I'm not familiar with the process by which we would need to register classes for yaml.

**Comment by trappitsch** [comment]
Well, I'm gonna play with this more. The current implementation is as unsafe/safe (depending on your standpoint) as before. Being able to register classes would definitely help this. Might have to play with some actual hardware to figure out why the test don't pass in my first tests with this.

**Comment by trappitsch** [comment]
So, looking at this in more details, it seems like we are not loading classes (which could be excepted from the safe loader), but we are loading module.name with `!!python/name` tags. The only way I can see to get this to work without restructuring a lot is to use `typ="unsafe"` in the loader. 

We have the following comment in the docstring of `config.load_instruments`:
```
    .. warning::
        The configuration file must be trusted, as the class name references
        allow for executing arbitrary code. Do not load instruments from
        configuration files sent over network connections.

        Note that keys in sections excluded by the ``conf_path`` argument are
        still processed, such that any side effects that may occur due to
        such processing will occur independently of the value of ``conf_path``.
```

The previous version with the older `ruamel.yaml` used unsafe loading by default.

**Comment by trappitsch** [comment]
Added some tests to fully cover the `config.load_instruments`. However, submitting to codecov seems to be failing...

**Comment by scasagrande** [comment]
As far as I can tell, v2 is still supported, I only see reports that v1 doesn't work anymore as of 2022.

**Comment by trappitsch** [comment]
let's see if it's just flaky...

**Comment by trappitsch** [comment]
Here we go :)

**Pull request merged**